test/loadtest.js

```diff
@@ -0,0 +1,92 @@
+'use strict'
+
+const http = require('http')
+const {test} = require('tap')
+const {createLogger} = require('../index.js')
+
+const TOTAL_LINES = 100000
+let httpCalls = 0
+let linesReceived = 0
+let sendEvents = 0
+let errorEvents = 0
+
+const server = http.createServer((req, res) => {
+  let buffer = ''
+  req.on('data', (data) => {
+    buffer += data
+  })
+  req.on('end', () => {
+    const body = JSON.parse(buffer)
+    if (++httpCalls % 2 === 0) {
+      res.statusCode = 500
+      res.end('POOF - The request should be retried')
+      return
+    }
+    linesReceived += body.ls.length
+    res.statusCode = 200
+    res.end('Ingested')
+  })
+})
+
+test('Load test to ensure no data loss and expected payloads', async (t) => {
+  const line = 'x'.repeat(500)
+
+  t.test('Start a local HTTP server', (tt) => {
+    server.listen(0)
+    server.on('listening', tt.end)
+  })
+
+  t.test(`Test ${TOTAL_LINES} lines`, (tt) => {
+    const logger = createLogger('<YOUR KEY HERE>', {
+      retryTimes: 100
+    , baseBackoffMs: 100
+    , maxBackoffMs: 500
+    , url: `http://localhost:${server.address().port}`
+    })
+
+    logger.on('send', (evt) => {
+      sendEvents++
+      tt.match(evt, {
+        httpStatus: 200
+      , firstLine: line
+      , lastLine: line
+      , totalLinesSent: Number
```

Comment:
Couldn't we make absolute assertions on these values?

Reply:
Not really because it's all asynchronous.  TBH, we don't need to do that here because the focus here is data preservation (no lost lines), not what the event values are.  Those are handled in other tests.